### PR TITLE
Make unique the output using -u param

### LINKED ISSUE #6: Parameter -u

### PR DESCRIPTION
**Pull request by yhabteab**
resolves #6

### CODE REVIEW

File: sort.php [hunk 1]
```diff
@@ -63,8 +63,14 @@ function bubbleSort($sort_array, $checkEquality)
 
 $bubbleSort = bubbleSort(checkForNextLine(file($fileName)), $funcCallback);
 
+if (key_exists("u", $options)) {
+    $makeUnique = array_unique($bubbleSort);
+} else {
+    $makeUnique = $bubbleSort;
+}
```
Comment (Al2Klimov) [inline sort.php]: Why not just assigning back to $bubbleSort, so you don't need the else?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: sort.php [hunk 2]
```diff
@@ -62,6 +62,9 @@ function bubbleSort($sort_array, $checkEquality)
 }
 
 $bubbleSort = bubbleSort(checkForNextLine(file($fileName)), $funcCallback);
+if (key_exists("u", $options)) {
+    $bubbleSort = array_unique($bubbleSort);
+}
```
Comment (Al2Klimov) [inline sort.php]: Use array_unique() before sorting, so you have less to sort.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: sort.php [hunk 3]
```diff
@@ -61,8 +61,13 @@ function bubbleSort($sort_array, $checkEquality)
     $fileName = 'php://stdin';
 }
 
-$bubbleSort = bubbleSort(checkForNextLine(file($fileName)), $funcCallback);
+$makeUnique = checkForNextLine(file($fileName));
+if (key_exists("u", $options)) {
+    $makeUnique = array_unique($makeUnique);
+    $makeUnique = array_values($makeUnique);
```
Comment (Al2Klimov) [inline sort.php]: Any way to shorten this into a one-liner?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [ ] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [ ] `git commit --amend --no-edit`
* [ ] `git push -f origin feature/sort-#6`

**Comment by yhabteab** [comment]
##Testprotocol #6

Today is a special day because we released two test protocols in one day. We are fantastic at it.
Let's go one-to-one through the previous published tests. We have recently released ``--numeric-sort`` and  ``--output`` and this time it's the ``--unique`` parameter.

The content of the in.txt file looks like this.
```
4
9
8
1
6
6
2
8
7
7
```
``php sort.php -u in.txt`` AND ``php sort.php -u < in.txt``  output:
```
1                                1
2                                2
4                                4
6                                6
7                                7
8                                8
9                                9
```
And the output of the php sort standard func is...
``sort -u in.txt`` AND ``sort -u < in.txt``:
```
1                                1
2                                2
4                                4
6                                6
7                                7
8                                8
9                                9
```

Have fun and see you next time!

Yonas :+1:

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**